src/flatdirectory.js

```diff
@@ -319,16 +452,15 @@ export class FlatDirectory {
                 const status = await this.#uploadBlob(fileContract, key, hexName, blobArr, chunkIdArr, chunkSizeArr, cost);
                 if (!status) {
                     cb.onFail(new Error("FlatDirectory: Sending transaction failed."));
```

Comment:
I wonder if onFail() can reveal more information such as at which chunk the upload failed, the total cost so far, etc?